### ISSUE TITLE
gsh wait issue

### ISSUE THREAD

**Issue by ksherlock** [ksherlock/gno-kern#8]
often on logging in, GSH will wait for a child process before the child process is enqueued (how this happens is another issue...)

1. Child process dies [?]
2. Kernel blocks further `SIGCHILD` signals and calls `SIGCHILD` signal handler
3. GSH signal handler calls `wait()` to reap the status.  but the waitq is empty
4. SIGCHILD signal is blocked so it doesn't get queued and wait() will never return.
5. hang

**Comment by ksherlock** [comment]
6598938013758874c30b271b2837b920ff33d3a2

**Comment by ksherlock** [comment]
Upon further investigation, this is the same double-wait issue as the other, with different effects due to the batch file.

1. when executing commands from a file (like gshrc)
2. fork creates pid 13
3. pid 13 quickly exits
4. kill(13, 0) to check if it's active
5. hey!, it's already dead, wait on it
6. fork creates pid 14
7. sigchild handler called (for pid 13 which has already been reaped)
8. sigchild handler calls wait()
9. pid 14 exits

The kernel temporarily blocked the SIGCHILD signal while executing the SIGCHILD handler,
which previously prevented pid 14 from being collected by `wait` from within the handler.
